t/t-credentials.sh

```diff
@@ -4,6 +4,9 @@
 
 ensure_git_version_isnt $VERSION_LOWER "2.3.0"
 
+export CREDSDIR="$REMOTEDIR/creds-credentials"
+setup_creds
```

Comment:
I confess to being a bit puzzled here because the commit description indicates the goal is to create a per-test credentials directory, but wouldn't this create a directory shared among all the tests in `t/t-credentials.sh`, and if I understand the problem there are several tests in that test suite which end up conflicting with each other, so this might not fully resolve the issue?  Quite possibly I'm missing some important detail here.

Reply:
Sorry, by per-test I mean test file, not `begin_test`/`end_test`.  Thus, the goal is to prevent the `t/t-credential.sh` and `t/t-clone.sh` tests from conflicting with each other.  The sets of tests in each file are fine on their own, just not together.